content/blog/2017/2017-02-14-declarative-notifications.adoc

```diff
@@ -0,0 +1,381 @@
+---
+layout: post
+title: "Declarative Pipeline: Notifications and Shared Libraries"
+tags:
+- tutorial
+- pipeline
+- declarative
+- plugins
+- notifications
+- slack
+- hipchat
+- emailext
+author: lnewman
+---
+
+NOTE: This is a guest post by link:https://github.com/bitwiseman[Liam Newman],
+Technical Evangelist at link:https://cloudbees.com[CloudBees].
+
+**Declare Your Pipelines!**
+link:/blog/2017/02/03/declarative-pipeline-ga/[Declarative Pipeline 1.0 is here]!
+This is the third post in a series showing some of the cool features of
+link:/doc/book/pipeline/syntax/#declarative-pipeline[Declarative Pipeline].
+
+
+In the
+link:/blog/2017/02/10/declarative-html-publisher/[previous post],
+we converted a Scripted Pipeline to a Declarative Pipeline, adding descriptive stages
+and `post` sections.  In one of those `post` blocks, we included a placeholder for
+sending notifications.
+
+In this blog post, we'll pull in the notification method we created for
+link:/blog/2016/07/18/pipline-notifications/[Sending Notifications in Pipeline],
+integrating notification services Slack, HipChat, and Email, into our Declarative Pipeline.
+Then we'll move that method into a shared library,
+keeping our `Jenkinsfile` concise and understandable,
+and letting use call that method from other projects.
+
+== Setup
+
+The setup for this post is a almost the same as the
+link:/blog/2017/02/10/declarative-html-publisher/[my previous Declarative Pipeline post].
+I've used a new branch in
+link:https://github.com/bitwiseman/hermann[my fork] of the
+link:https://github.com/reiseburo/hermann[Hermann project]:
+link:https://github.com/bitwiseman/hermann/tree/blog/declarative/notifications[`blog/add-declarative/notifications`].
```

Comment:
Minor: the anchor text should match the URL, i.e. `add-declarative` -> `declarative`